### PR TITLE
Fuse the add of two convolutions

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: test/simplify_algebra_test.cpp [hunk 1]
```diff
@@ -262,4 +262,127 @@ TEST_CASE(simplify_inner_broadcast)
...
+    auto sum   = p.add_instruction(migraphx::op::add{}, conv1, conv2);
+    p.add_instruction(pass_op{}, sum);
+    auto s = p.get_shape();
+    p.compile(simplify_algebra_target{});
+    EXPECT(s == p.get_shape());
+    EXPECT(std::count_if(
+               p.begin(), p.end(), [](auto&& ins) { return ins.name() == "convolution"; }) == 1);
+}
+
+TEST_CASE(simplify_add_conv2)
```
Comment (kahmed10) [inline test/simplify_algebra_test.cpp]: can you rewrite the names of the test cases to clarify what each one is testing?

File: test/gpu/ops_test.cpp [hunk 2]
```diff
@@ -843,6 +843,44 @@ struct test_conv_bias_clipped_relu : verify_program<test_conv_bias_clipped_relu>
...
+            p.add_literal(migraphx::generate_literal({migraphx::shape::float_type, {2, 8, 3, 3}}));
+        auto conv1 = p.add_instruction(migraphx::op::convolution{}, x, w);
+        auto conv2 = p.add_instruction(migraphx::op::convolution{}, y, v);
+        auto sum   = p.add_instruction(migraphx::op::add{}, conv1, conv2);
+        p.add_instruction(migraphx::op::exp{}, sum);
+        return p;
+    }
+};
+
+struct test_conv_add2 : verify_program<test_conv_add2>
```
Comment (kahmed10) [inline test/gpu/ops_test.cpp]: can you rewrite the test name to something like "test_conv_add_1x1_diff_strides"

File: test/simplify_algebra_test.cpp (hunk 1 above)
Comment (pfultz2) [reply test/simplify_algebra_test.cpp]: Done

File: test/gpu/ops_test.cpp (hunk 2 above)
Comment (pfultz2) [reply test/gpu/ops_test.cpp]: Done

File: src/include/migraphx/op/as_shape.hpp [hunk 3]
```diff
@@ -29,7 +29,7 @@ struct as_shape
     shape compute_shape(const std::vector<shape>& inputs) const
     {
         check_shapes{inputs, *this}.has(1).standard();
-        assert(inputs.front().elements() == s.elements());
+        assert(inputs.front().elements() >= s.elements());
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply src/include/migraphx/op/as_shape.hpp]: Yes, it will skip over data, but that is what we want.

**Pull request merged**